File: python_programs/binary_tree.py
class Node:
		def __init__(self,val,left,right):
			self.val = val
			self.left = left
			self.right = right
class binary_tree:
	def __init__(self):
		self.root = None
		self.root = self.add(self.root)

	def get_root(self):
		return self.root

	def add(self,parent):
		if(parent == None):
			print('enter value?:')
			val = int(input())
			nn = Node(val , None , None)
			parent = nn
		print('left child??')
		lc = input()
		if lc == 'yes':
			parent.left = self.add(parent.left)
		print('right child?')
		rc = input()
		if rc == 'yes':
			parent.right = self.add(parent.right)
		return parent

	def display(self,parent):
		if parent == None:
			return
		print(parent.val)
		self.display(parent.left)
		self.display(parent.right)

tree = binary_tree()
tree.display(tree.get_root())

File: cpp/binary_tree.cpp
#include<bits/stdc++.h>
using namespace std;

struct Node
{
	int val;
	Node* left;
	Node* right;
};
Node* root = NULL;

Node* add(Node* parent){
	if(parent == NULL){
		Node*nn= new Node();
		cout<<"enter the value:\n";
		int value;
		cin>>value;
		nn->val = value;
		parent = nn;
	}

	string lc;
	cout<<"left child?? :";
	cin>>lc;

	if(lc == "yes")
		parent->left = add(parent->left);
	string rc;
	cout<<"right child?? :";
	cin>>rc;
	if(rc == "yes")
		parent->right = add(parent->right);
	return parent;
}

void display(Node* parent){
	if(parent == NULL)
		return;
	cout<<parent->val<<endl;
	display(parent->left);
	display(parent->right);
}
int main(int argc, char const *argv[])
{
	root = add(root);
	display(root);
	return 0;
}